applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1252,13 +1253,34 @@ static void unicast_discovery_complete_cb(struct bt_conn *conn, int err,
 					  const struct bt_csip_set_coordinator_set_member *member,
 					  const struct bt_csip_set_coordinator_csis_inst *csis_inst)
 {
-	if (err) {
+	int ret;
+	struct le_audio_msg msg;
+	uint8_t index;
+
+	ret = device_index_get(conn, &index);
+	if (ret) {
+		return;
+	}
+
+	if (err || csis_inst == NULL) {
```

Comment:
Check for err at the beginning of this function. If there is an err, it may be impossible to do `device_index_get`

Reply:
No, the err usually indicates that the discovery didn't complete successfully, meaning potentially that the device doesn't have CAS. In can still be connected, especially after this PR is merged and upmerged: https://github.com/zephyrproject-rtos/zephyr/pull/73475

Reply:
As discussed device_index_get should always return an index here since the device has been added to the unicast_servers array in the function that started the discovery process.